trieste/utils/pareto.py

```diff
@@ -105,3 +105,32 @@ def _bounds_2d(front: TensorType) -> BoundedVolumes:
         upper = tf.concat([range_ + 1, pf_ext_idx[::-1, 1:][: pf_ext_idx[-1, 0]]], axis=-1)
 
         return BoundedVolumes(lower, upper)
+
+    def hypervolume_indicator(self, reference: tf.Tensor) -> tf.Tensor:
+        """
+        Method to calculate the hypervolume indicator
+        The hypervolume indicator is the volume of the dominated region.
+
+        :param reference: a reference point to use, with shape [D].
```

Comment:
could you add an assert_shapes that checks the shape of this - particularly that D matches the `front` or `bounds`

Reply:
done